hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java

```diff
@@ -97,7 +97,7 @@ protected BaseTableMetadata(HoodieEngineContext engineContext, HoodieMetadataCon
     this.isMetadataTableInitialized = dataMetaClient.getTableConfig().isMetadataTableAvailable();
 
     if (metadataConfig.enableMetrics()) {
-      this.metrics = Option.of(new HoodieMetadataMetrics(Registry.getRegistry("HoodieMetadata")));
+      this.metrics = Option.of(new HoodieMetadataMetrics(HoodieMetricsConfig.newBuilder().fromProperties(metadataConfig.getProps()).build()));
```

Comment:
metadataConfig is not going to contain any metrics related props. this is on the reader side. 
What we have fixed in HoodieMetadataWriteUtils is applicable for metadata writer and not reader. 
We need some fixes here. if not, the metrics related props may not be carried over to this code snippet.

Reply:
but I am not sure if we can even get that. bcoz, the query engine is not going to set any writer props (for eg metrics related ones). So, its not feasible for us to instantiate this properly on the reader side :(